### PR TITLE
Use unique fake words for attribute namespaces

### LINKED ISSUE #7204: Flickering test ./spec/bootstrap/features/webui/attributes_spec.rb:52
As we don't reset attribute names for rspec (due to some of them required as seeds), we run into test data collusion.

I can reproduce the problem by running:

```
while rspec ./spec/bootstrap/features/webui/attributes_spec.rb:52; do echo; done
```

Resulting in:

```
Failures:

  1) Bootstrap_Attributes with an attribute for a project with a package add attribute with values
     Failure/Error: expect(page).to have_content('Attribute was successfully created.')
       expected to find text "Attribute was successfully created." in "AttribType::UnknownAttributeTypeError in Webui::AttributeController#edit\nAttribute Type perferendis:MyImageTemplates does not exist\nExtracted source (around line #49):\n47\n48\n49\n50\n51\n52\n              \n    ats = joins(:attrib_namespace).where('attrib_namespaces.name = ? and attrib_types.name = ?', namespace, name)\n    if or_fail && ats.count != 1\n      raise UnknownAttributeTypeError, \"Attribute Type \#{namespace}:\#{name} does not exist\"\n    end\n    ats.first\n  end\nRails.root: /obs/src/api\nApplication Trace | Framework Trace | Full Trace\napp/models/attrib_type.rb:49:in `find_by_namespace_and_name'\napp/models/attrib_type.rb:36:in `find_by_name'\napp/models/attrib_type.rb:28:in `find_by_name!'\napp/models/attrib.rb:41:in `find_by_container_and_fullname'\napp/controllers/webui/attribute_controller.rb:32:in `edit'\nconfig/initializers/wrap_parameters.rb:38:in `call'\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\nRequest\nParameters:\n{\"project\"=>\"home:user_1\", \"package\"=>\"package_1\", \"attribute\"=>\"perferendis:MyImageTemplates\"}\nToggle session dump\nToggle env dump\nResponse\nHeaders:\nNone". (However, it was found 1 time including non-visible text.)
     # ./spec/support/features/features_attribute.rb:8:in `add_attribute_with_values'
     # ./spec/bootstrap/features/webui/attributes_spec.rb:55:in `block (4 levels) in <top (required)>'
```

And as the log file shows that attrib name space was used before:

```
frontend@8b0cb4c29e3d:/obs/src/api> grep perferendis log/test.log
[17368:2.71]   AttribNamespace Create (0.3ms)  INSERT INTO `attrib_namespaces` (`name`) VALUES ('perferendis')
[fe4a4065-60df-4a2c-8a60-aa23c6f58572] [17368:0.04] Redirected to http://frontend:3005/attribs/home:user_1/package_1/perferendis:MyImageTemplates/edit
[80f68eca-8138-4b29-af41-2b1c260a4f21] [17368:0.04] Started GET "/attribs/home:user_1/package_1/perferendis:MyImageTemplates/edit" for 172.23.0.6 at 2019-03-15 08:19:46 +0000
[80f68eca-8138-4b29-af41-2b1c260a4f21] [17368:0.04]   Parameters: {"project"=>"home:user_1", "package"=>"package_1", "attribute"=>"perferendis:MyImageTemplates"}
[80f68eca-8138-4b29-af41-2b1c260a4f21] [17368:0.05]    (0.3ms)  SELECT COUNT(*) FROM `attrib_types` INNER JOIN `attrib_namespaces` ON `attrib_namespaces`.`id` = `attrib_types`.`attrib_namespace_id` WHERE (attrib_namespaces.name = 'perferendis' and attrib_types.name = 'MyImageTemplates')
[80f68eca-8138-4b29-af41-2b1c260a4f21] [17368:0.05]   AttribType Load (0.3ms)  SELECT  `attrib_types`.* FROM `attrib_types` INNER JOIN `attrib_namespaces` ON `attrib_namespaces`.`id` = `attrib_types`.`attrib_namespace_id` WHERE (attrib_namespaces.name = 'perferendis' and attrib_types.name = 'MyImageTemplates') ORDER BY `attrib_types`.`id` ASC LIMIT 1
[5e264197-871f-4de2-a0b1-735ca580c676] [17368:3.93] Redirected to http://frontend:3005/attribs/home:user_1/package_1/perferendis:MyImageTemplates/edit
[ee797f90-baec-488c-9fc9-aa7fe218df10] [17368:3.93] Started GET "/attribs/home:user_1/package_1/perferendis:MyImageTemplates/edit" for 172.23.0.6 at 2019-03-15 08:19:47 +0000
[ee797f90-baec-488c-9fc9-aa7fe218df10] [17368:3.94]   Parameters: {"project"=>"home:user_1", "package"=>"package_1", "attribute"=>"perferendis:MyImageTemplates"}
[ee797f90-baec-488c-9fc9-aa7fe218df10] [17368:3.94]    (0.3ms)  SELECT COUNT(*) FROM `attrib_types` INNER JOIN `attrib_namespaces` ON `attrib_namespaces`.`id` = `attrib_types`.`attrib_namespace_id` WHERE (attrib_namespaces.name = 'perferendis' and attrib_types.name = 'MyImageTemplates')
[ee797f90-baec-488c-9fc9-aa7fe218df10] [17368:3.94]   AttribType Load (0.3ms)  SELECT  `attrib_types`.* FROM `attrib_types` INNER JOIN `attrib_namespaces` ON `attrib_namespaces`.`id` = `attrib_types`.`attrib_namespace_id` WHERE (attrib_namespaces.name = 'perferendis' and attrib_types.name = 'MyImageTemplates') ORDER BY `attrib_types`.`id` ASC LIMIT 1
[17416:2.61]   AttribNamespace Create (16.2ms)  INSERT INTO `attrib_namespaces` (`name`) VALUES ('perferendis')
[ed5f3c4a-20ee-4e34-9785-c7891eeb6560] [17416:0.04] Redirected to http://frontend:3005/attribs/home:user_1/package_1/perferendis:MyImageTemplates/edit
[47d14b1a-27f0-4143-8f85-0da9ada4fd8f] [17416:0.04] Started GET "/attribs/home:user_1/package_1/perferendis:MyImageTemplates/edit" for 172.23.0.6 at 2019-03-15 08:20:14 +0000
[47d14b1a-27f0-4143-8f85-0da9ada4fd8f] [17416:0.04]   Parameters: {"project"=>"home:user_1", "package"=>"package_1", "attribute"=>"perferendis:MyImageTemplates"}
[47d14b1a-27f0-4143-8f85-0da9ada4fd8f] [17416:0.05]    (0.2ms)  SELECT COUNT(*) FROM `attrib_types` INNER JOIN `attrib_namespaces` ON `attrib_namespaces`.`id` = `attrib_types`.`attrib_namespace_id` WHERE (attrib_namespaces.name = 'perferendis' and attrib_types.name = 'MyImageTemplates')
[47d14b1a-27f0-4143-8f85-0da9ada4fd8f] [17416:0.05] AttribType::UnknownAttributeTypeError (Attribute Type perferendis:MyImageTemplates does not exist):
```

Faker words are a finite resource :)

I'm afraid the only cure to this is actually resetting attribute namespaces as well and recreating the seed data in the tests requiring them (or e.g. in all feature tests).

### PR DESCRIPTION
**Pull request by coolo**
Fixes #7204

### CODE REVIEW

**Comment by coolo** [comment]
And sorry for the PR storm this morning - I shouldn't have taken the coffee.

File: src/api/spec/support/faker.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+RSpec.configure do |config|
+  config.before do
+    # avoid running into https://github.com/openSUSE/open-build-service/issues/7204
```
Comment (Ana06) [inline src/api/spec/support/faker.rb]: I would shortly explaining instead of linking a GitHub issue

**Comment by coolo** [comment]
This is IMO more a workaround - I would actually prefer if every example started on a clean database and we injected standard database entries per hook. Any best practises known for this kind of problem?

File: src/api/spec/factories/attrib_namespaces.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 FactoryBot.define do
   factory :attrib_namespace do
-    name { Faker::Lorem.word }
+    name { Faker::Lorem.unique.word }
```
Comment (bgeuken) [inline src/api/spec/factories/attrib_namespaces.rb]: JFYI, you could also use a [sequence](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#sequences) here

**Comment by bgeuken** [comment]
> coolo commented 3 days ag
> This is IMO more a workaround - I would actually prefer if  every example started on a clean database and we injected standard  database entries per hook. Any best practises known for this kind of  problem?

We use [database cleaner](https://github.com/DatabaseCleaner/database_cleaner) for that.
But for some of the models (that store general server configuration) we skip that, see [spec/support/database_cleaner.rb](https://github.com/openSUSE/open-build-service/blob/master/src/api/spec/support/database_cleaner.rb#L5).

File: src/api/spec/factories/attrib_namespaces.rb (hunk 2 above)
Comment (coolo) [reply src/api/spec/factories/attrib_namespaces.rb]: I don't think so. I believe the sessions are per suite - which make them even more dangerous to collision on test reruns.

**Comment by coolo** [comment]
> > coolo commented 3 days ag
> > This is IMO more a workaround - I would actually prefer if  every example started on a clean database and we injected standard  database entries per hook. Any best practises known for this kind of  problem?
> 
> We use [database cleaner](https://github.com/DatabaseCleaner/database_cleaner) for that.
> But for some of the models (that store general server configuration) we skip that, see [spec/support/database_cleaner.rb](https://github.com/openSUSE/open-build-service/blob/master/src/api/spec/support/database_cleaner.rb#L5).

Please tell me you actually read #7204 before commenting.

**Comment by coolo** [comment]
went with 7208

**Pull request closed without merging**